### PR TITLE
Observable check Hermitian

### PR DESCRIPTION
**Pull request by SamDuffield**
Clarify `observable.py` to only accept Hermitian matrices/tensors

### CODE REVIEW

File: qujax/observable.py [hunk 1]
```diff
@@ -4,52 +4,62 @@
...
 
-def _statetensor_to_single_expectation_func(gate_tensor: jnp.ndarray,
-                                            qubit_inds: Sequence[int]) -> Callable[[jnp.ndarray], float]:
+
+def statetensor_to_single_expectation(statetensor: jnp.ndarray,
+                                      hermitian: jnp.ndarray,
+                                      qubit_inds: Sequence[int]) -> float:
     """
-    Creates a function that maps statetensor to its expected value under the given gate unitary and qubit indices.
+    Evaluates expected value of statetensor through a Hermitian matrix (in tensor form).
```
Comment (gamatos) [inline qujax/observable.py:17]:
```suggestion
    Evaluates expectation value of an observable represented by a Hermitian matrix (in tensor form).
```

File: qujax/observable.py [hunk 2]
```diff
@@ -4,52 +4,62 @@
...
 
-def get_statetensor_to_expectation_func(gate_seq_seq: Sequence[Sequence[Union[str, jnp.ndarray]]],
+def get_statetensor_to_expectation_func(hermitian_seq_seq: Sequence[Sequence[Union[str, jnp.ndarray]]],
                                         qubits_seq_seq: Sequence[Sequence[int]],
                                         coefficients: Union[Sequence[float], jnp.ndarray]) \
         -> Callable[[jnp.ndarray], float]:
     """
-    Converts gate strings (or arrays), qubit indices and coefficients into a function that
-    converts statetensor into expected value.
+    Converts strings (or arrays) representing Hermitian matrices, qubit indices and
```
Comment (gamatos) [inline qujax/observable.py:57]:
```suggestion
    Takes strings (or arrays) representing Hermitian matrices, qubit indices and
```

File: qujax/observable.py [hunk 3]
```diff
@@ -4,52 +4,62 @@
...
+    coefficients into a function that converts a statetensor into an expected value.
```
Comment (gamatos) [inline qujax/observable.py:58]:
```suggestion
   a list of coefficients and returns a function that converts a statetensor into an expected value.
```

File: qujax/observable.py [hunk 4]
```diff
@@ -4,52 +4,62 @@
...
 
     Args:
-        gate_seq_seq: Sequence of sequences of gates.
-            Each gate is either a tensor (jnp.ndarray) or a string corresponding to an array in qujax.gates.
+        hermitian_seq_seq: Sequence of sequences of Hermitian matrices/tensors.
```
Comment (gamatos) [inline qujax/observable.py:61]:
```suggestion
        hermitian_seq_seq: Sequence of sequences of Hermitian matrices (in tensor form).
```

File: qujax/observable.py [hunk 5]
```diff
@@ -4,52 +4,62 @@
...
+            Each Hermitian is either a tensor (jnp.ndarray) or a string in ('X', 'Y', 'Z').
```
Comment (gamatos) [inline qujax/observable.py:62]:
```suggestion
            Each Hermitian matrix is either represented by a tensor (jnp.ndarray) or by a list of 'X', 'Y' or 'Z' characters corresponding to the standard Pauli matrices.
```

File: qujax/observable.py [hunk 6]
```diff
@@ -59,28 +69,30 @@ def get_statetensor_to_expectation_func(gate_seq_seq: Sequence[Sequence[Union[st
         Function that takes statetensor and returns expected value (float).
     """
 
-    def get_gate_tensor(gate_seq: Sequence[Union[str, jnp.ndarray]]) -> jnp.ndarray:
+    def get_hermitian_tensor(hermitian_seq: Sequence[Union[str, jnp.ndarray]]) -> jnp.ndarray:
         """
-        Convert sequence of gate strings into single gate unitary (in tensor form).
+        Convert sequence of Hermitian strings/arrays into single array (in tensor form).
```
Comment (gamatos) [inline qujax/observable.py:74]:
```suggestion
        Convert a sequence of observables represented by Pauli strings or Hermitian matrices in tensor form into single array (in tensor form).
```

File: qujax/observable.py (hunk 4 above)
Comment (SamDuffield) [reply qujax/observable.py:61]: here they don't actually have to be in tensor form as they will be reshaped anyway

**Review by gamatos — APPROVED** [review]

**Pull request merged**